snforge_std/src/cheatcodes/events.cairo

```diff
@@ -102,10 +151,16 @@ fn assert_if_emitted(ref self: EventSpy, event: Event) -> bool {
             break false;
         }
 
-        if event_name_hash(event.name) == *self.events.at(j).name
-            && event.from == *emitted_events.at(j).from
-            && @event.keys == emitted_events.at(j).keys
-            && @event.data == emitted_events.at(j).data {
+        let emitted = match event.clone() {
+            Event::Named(named_event) => {
+                let mut cloned_event = named_event.clone();
+                cloned_event.name = event_name_hash(cloned_event.name);
+                Event::Named(cloned_event) == emitted_events.at(j).clone()
+            },
+            Event::Unnamed(unnamed_event) => Event::Unnamed(unnamed_event) == emitted_events.at(j).clone()
```

Comment:
```suggestion
           unnamed => unnamed == emitted_events.at(j).clone()
```

Reply:
We have to match `Event::Unnamed` and then compare it. `unnamed` is not a variant